### DISCUSSION TITLE
Get Occurrences based on a RRULE FREQ=SECONDLY

### DISCUSSION THREAD

**Discussion by DaleyKD** [ical-org/ical.net#615] [Q&A]
Does this library support getting occurrences if we use a secondly recurrence pattern?

Our project doesn't use this for calendar events, but more to help us determine when to run the next job (as part of our job scheduler). And SOME jobs need to run every 45 seconds or so.

I've been trying to get it to work, but have had no luck. It seems to work for FREQ=MINUTELY.

**Comment by axunonb** [comment]
@DaleyKD
We took over this project few days ago and are heavily working on all kinds of `RRULE`s. Please allow some time to get on top of things pending.
In the meantime maybe post the sample code that doesn't seem to work.

@minichma 

`Evaluator` contains `FREQ=SECONDLY`
https://github.com/ical-org/ical.net/blob/a0179f811b425fe5bde004c102da209907681a8f/Ical.Net/Evaluation/Evaluator.cs#L62-L64
and `RecurrencePattern` contains comments for a fixed bug
https://github.com/ical-org/ical.net/blob/a0179f811b425fe5bde004c102da209907681a8f/Ical.Net/DataTypes/RecurrencePattern.cs#L91

Is RRULE FREQ=SECONDLY on our todo list?

**Reply by minichma** [reply]
I think, secondly should work, but it's disabled by default. There's a flag to control this.

https://github.com/ical-org/ical.net/blob/a0179f811b425fe5bde004c102da209907681a8f/Ical.Net/Calendar.cs#L169

**Comment by axunonb** [comment]
```C#
[Test]
public void SecondlyRecurrence()
{
    var dateTime = DateTime.UtcNow;

    // Create a new calendar
    var calendar = new Calendar
    {
        RecurrenceRestriction = RecurrenceRestrictionType.NoRestriction
    };

    var calendarEvent = new CalendarEvent
    {
        Summary = "Recurring Event Every 45 Seconds",
        DtStart = new CalDateTime(dateTime),
        DtEnd = new CalDateTime(dateTime.AddHours(1)),
        RecurrenceRules = new List<RecurrencePattern>
        {
            new RecurrencePattern(FrequencyType.Secondly, 45)
        }
    };

    // Add the event to the calendar
    calendar.Events.Add(calendarEvent);

    // Serialize the calendar to an iCalendar string
    var serializer = new CalendarSerializer();
    var serializedCalendar = serializer.SerializeToString(calendar);

    // Evaluate the recurrences
    var occurrences = calendar.GetOccurrences(dateTime, dateTime.AddHours(1));
    // or
    // var occurrences = calendarEvent.GetOccurrences(dateTime, dateTime.AddHours(1));

    Assert.That(occurrences, Has.Count.EqualTo(80));
}
```
Serializes correctly:
```
BEGIN:VCALENDAR
PRODID:-//github.com/ical-org/ical.net//NONSGML ical.net 4.0//EN
VERSION:2.0
X-DDAY-ICAL-RECURRENCE-RESTRICTION:NoRestriction
BEGIN:VEVENT
DTEND:20241022T232551Z
DTSTAMP:20241022T222551Z
DTSTART:20241022T222551Z
RRULE:FREQ=SECONDLY;INTERVAL=45
SEQUENCE:0
SUMMARY:Recurring Event Every 45 Seconds
UID:888078c5-ff36-46cd-b16c-a6e1609e8c01
END:VEVENT
END:VCALENDAR
```

Fails here when calling `GetOccurrences`.
https://github.com/ical-org/ical.net/blob/a0179f811b425fe5bde004c102da209907681a8f/Ical.Net/Evaluation/RecurrencePatternEvaluator.cs#L181-L191
because variable `evaluationRestriction` is `RecurrenceRestrictionType.Default` although set with `RecurrenceRestrictionType.NoRestriction`.

**Comment by axunonb** [comment]
Closed in favor of #622
